web/pdf_document_properties.js

```diff
@@ -87,71 +113,71 @@ class PDFDocumentProperties {
    * @param {Object} pdfDocument - A reference to the PDF document.
    * @param {string} url - The URL of the document.
    */
-  setDocumentAndUrl(pdfDocument, url) {
+  setDocument(pdfDocument, url) {
+    if (this.pdfDocument) {
+      this._reset();
+    }
+    if (!pdfDocument) {
+      return;
+    }
     this.pdfDocument = pdfDocument;
-    this.url = url;
+
+    this._url = url;
     this._dataAvailableCapability.resolve();
   }
 
   /**
-   * @private
+   * Set the file size of the PDF document. This method is used to
+   * update the file size in the document properties overlay once it
+   * is known so we do not have to wait until the entire file is loaded.
+   *
+   * @param {number} fileSize - The file size of the PDF document.
    */
-  _getProperties() {
-    if (!OverlayManager.active) {
-      // If the dialog was closed before `_dataAvailableCapability` was
-      // resolved, don't bother updating the properties.
-      return;
+  setFileSize(fileSize) {
+    if (typeof fileSize === 'number' && fileSize > 0) {
+      this._rawFileSize = fileSize;
     }
-    // Get the file size (if it hasn't already been set).
-    this.pdfDocument.getDownloadInfo().then((data) => {
-      if (data.length === this.rawFileSize) {
-        return;
-      }
-      this.setFileSize(data.length);
-      this._updateUI(this.fields['fileSize'], this._parseFileSize());
-    });
+  }
 
-    // Get the document properties.
-    this.pdfDocument.getMetadata().then((data) => {
-      var content = {
-        'fileName': getPDFFileNameFromURL(this.url),
-        'fileSize': this._parseFileSize(),
-        'title': data.info.Title,
-        'author': data.info.Author,
-        'subject': data.info.Subject,
-        'keywords': data.info.Keywords,
-        'creationDate': this._parseDate(data.info.CreationDate),
-        'modificationDate': this._parseDate(data.info.ModDate),
-        'creator': data.info.Creator,
-        'producer': data.info.Producer,
-        'version': data.info.PDFFormatVersion,
-        'pageCount': this.pdfDocument.numPages
-      };
-
-      // Show the properties in the dialog.
-      for (var identifier in content) {
-        this._updateUI(this.fields[identifier], content[identifier]);
+  /**
+   * @private
+   */
+  _reset() {
+    if (this.pdfDocument) {
+      // Reset the dialog when a new PDF file is opened (issue 8371).
+      for (let id in this.fields) {
+        this._updateField(this.fields[id]);
```

Comment:
You're right, but to me it just looked strange to update a field without providing a value. The fallback is implicit if you just read that line without going to the actual `updateField` method. Adding a comment above this line saying that the fallback value is used will indeed help to avoid confusion when reading this file.